### PR TITLE
feat: Added admin donation filter buttons

### PR DESCRIPTION
**Pull request by AmandaMisjuwar**
## Implementation description. How did you make this change?
* Added admin donation views export and filter buttons
* Added frontend filter functionality for admin donation view (gets all donations on first render and filter on FE)
* Fixed mobile + desktop spacing for donation and check in views

## Steps to test
1. Create two donations 
2. Create a volunteer account
3. Use PG Admin to assign a volunteer to one of the two donations by editing directly when viewing the table 
4. Log in as an admin
5. Go to Fridge Management > Scheduled Donations
6. All scheduled donations should show as default filter is All
7. Select filter by Filled
8. Only the scheduled donation with the assigned volunteer should show
9. Select filter by Unassigned
10. Only the scheduled donation with no assigned volunteers should show
11. Make sure mobile + desktop matches designs and check for the Check-In view as well

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages follow conventional commits and are descriptive. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s) 
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR
- [ ] The appropriate tests if necessary have been written

### CODE REVIEW

**Comment by AmandaMisjuwar** [comment]
I spent almost 1 hr trying to fix spacing to find the only thing i needed was `width=100%` D:<

File: frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx [hunk 1]
```diff
@@ -52,6 +86,25 @@ const ViewDonationsAndCheckIns = ({
...
+        (schedule) => schedule.volunteerId != null,
+      );
+    // selectedFilter === AccountFilterType.DONOR
+    else
+      filteredDonations = schedules.filter(
+        (schedule) => schedule.volunteerId == null,
+      );
+
+    setFilteredSchedules(filteredDonations);
+  }, [selectedFilter]);
```
Comment (hanlinc27) [inline frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx:106]: can we memoize this?
```suggestion
  React.useMemo(()=>{
    if (selectedFilter === DonationFilterType.FILLED){
      setFilteredSchedules([...schedules.filter((schedule) => schedule.volunteerId != null)]);
    }
    else if (selectedFilter === DonationFilterType.UNASSIGNED){
      setFilteredSchedules([...schedules.filter((schedule) => schedule.volunteerId == null)]);
    }
    else {
      setFilteredSchedules([...schedules]);
    }
  }, [selectedFilter]);
```

File: frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx [hunk 2]
```diff
@@ -69,31 +127,94 @@ const ViewDonationsAndCheckIns = ({
...
+              leftIcon={<DownloadIcon />}
+              flex={1}
+            >
+              Export
+            </Button>
+            <Select
+              color="hubbard.100"
+              background={isMobile ? "dorian.100" : "none"}
+              border={isMobile ? "none" : "1px"}
+              borderColor={isMobile ? "none" : "dorian.100"}
```
Comment (hanlinc27) [inline frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx:155]:
```suggestion
              background={["dorian.100","none"]}
              border={["none" : "1px"]}
              borderColor={["none" : "dorian.100]}
```

File: frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx [hunk 3]
```diff
@@ -69,31 +127,94 @@ const ViewDonationsAndCheckIns = ({
...
+              onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+                handleSelectFilter(e);
+              }}
+              flex={isMobile ? 1.5 : 2.5}
```
Comment (hanlinc27) [inline frontend/src/components/pages/AdminDashboard/ViewDonationsAndCheckIns.tsx:159]:
```suggestion
              flex={[1.5,2.5]}

```

**Review by hanlinc27 — CHANGES_REQUESTED** [review]
functionality seems to work for me! just a quick comment 💯

**Review by hanlinc27 — APPROVED** [review]

**Pull request merged**